ingest/processors/attach.go

```diff
@@ -0,0 +1,106 @@
+/*************************************************************************
+ * Copyright 2018 Gravwell, Inc. All rights reserved.
+ * Contact: <legal@gravwell.io>
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD 2-clause license. See the LICENSE file for details.
+ **************************************************************************/
+
+package processors
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gravwell/gcfg"
+	"github.com/gravwell/gravwell/v3/ingest/attach"
+	"github.com/gravwell/gravwell/v3/ingest/config"
+	"github.com/gravwell/gravwell/v3/ingest/entry"
+)
+
+const (
+	AttachProcessor string = `attach`
+)
+
+// AttachLoadConfig loads the configuration for the attach processor
+// It converts the VariableConfig to an attach.AttachConfig
+func AttachLoadConfig(vc *config.VariableConfig) (c attach.AttachConfig, err error) {
+	// The VariableConfig and AttachConfig have the same underlying structure
+	// (gcfg.Idxer and Vals map[gcfg.Idx]*[]string)
+	c.Idxer = vc.Idxer
+
+	// Copy the Vals map so we can filter without affecting the source config
+	c.Vals = make(map[gcfg.Idx]*[]string, len(vc.Vals))
+	for k, v := range vc.Vals {
+		c.Vals[gcfg.Idx(k)] = v
+	}
+
+	// Filter out the "type" key which is used for preprocessor selection
+	// but should not be attached as an enumerated value
+	delete(c.Vals, c.Idx("type"))
+
+	err = c.Verify()
```

Comment:
right here, if err == nil, swing through the map of c.Vals and check for a key of `$UUID`, if you see it, throw an error.

This way we can warn the user rather than push in useless bytes.